jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+import os
+
+class FileSystemCache:
+
+  def __init__(self, path: str):
+    """Sets up a cache at 'path'. Cached values may already be present."""
+    if not os.path.isdir(path):
```

Comment:
I think you meant to have
```
if not os.path.exists(path):
    os.makedirs(path)
```
An even better version might be

```
os.makedirs(path, exist_ok=True)
```

Reply:
I made this change, but actually could it pose a problem such as replicating a cache directory? Since I don't have an if statement now it says a directory will be made no matter what the path is but don't I just want it to make a directory if the path given isn't already one?

Reply:
os.makedirs("path/to/directory", exist_ok=True)  # succeeds even if directory exists.

see
https://stackoverflow.com/questions/273192/how-can-i-safely-create-a-nested-directory

this is equivalent do "mkdir -p"